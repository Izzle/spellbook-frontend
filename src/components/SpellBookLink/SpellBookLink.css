
.SpellBookLink__button {
	box-shadow:inset 0px 1px 1px 0px #a6827e;
	background:linear-gradient(to bottom, #553F30 5%, #634b30 100%);
	background-color:#553F30;
	border-radius:11px;
	border:1px solid #54381e;
	display:inline-block;
	cursor:pointer;
	color: white;
	font-family:Arial;
	font-size:20px;
	padding:32px 76px;
	text-decoration:none;
    text-shadow:0px 1px 0px #4d3534;
    margin-bottom: 24px;
}

.SpellBookLink__button:hover {
	background:linear-gradient(to bottom, #634b30 5%, #553F30 100%);
	background-color:#634b30;
}

.SpellBookLink__button:active {
	position:relative;
	top:1px;
}