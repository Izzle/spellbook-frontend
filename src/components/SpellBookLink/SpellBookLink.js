import React from 'react';
import { Link } from 'react-router-dom';
import './SpellBookLink.css';

export default function SpellBookLink(props){

    return(
        
      //<Link to={`/spellbook/${props.id}`} className='SpellBookLink__Link'>
        <Link to={{
          pathname: `/spellbook/${props.id}`,
          state: {
            spellbook_name: props.name
          }
        }} className='SpellBookLink__Link'>
          <button className='SpellBookLink__button'>{props.name}</button>
        </Link>
    );
}