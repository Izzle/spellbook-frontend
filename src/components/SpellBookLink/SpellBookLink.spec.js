import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import SpellBookLink from './SpellBookLink';
import { BrowserRouter } from 'react-router-dom';


describe('<SpellBookLink />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BrowserRouter><SpellBookLink /></BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders this UI as expected', () => {
        const tree = renderer.create(<BrowserRouter><SpellBookLink /></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });    
});