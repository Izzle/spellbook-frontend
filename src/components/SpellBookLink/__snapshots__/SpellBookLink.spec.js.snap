// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpellBookLink /> renders this UI as expected 1`] = `
<a
  className="SpellBookLink__Link"
  href="/spellbook/undefined"
  onClick={[Function]}
>
  <button
    className="SpellBookLink__button"
  />
</a>
`;
